File: c/hamming/src/hamming.c
#include "hamming.h"
#include <stddef.h>
#include <string.h>

int compute(const char *lhs, const char *rhs) {
    if (!lhs || !rhs || !(strlen(lhs) == strlen(rhs)))
        return -1;

    int distance = 0;
    int l = (int)strlen(lhs);
    for(int i=0; i<l; i++)
        if(lhs[i] != rhs[i])
            distance++;
    return distance;
}

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
    if #a ~= #b then return -1 end
    local d = 0
    for i=1,#a do
        if a:sub(i,i) ~= b:sub(i,i) then
            d = d + 1
        end
    end
    return d
end

return Hamming
